File: software engineering-back/internal/model/dto/response/quiz.go
package response

// QuizResponse 答题记录响应
type QuizResponse struct {
	QuizID             uint             `json:"quiz_id"`              // 答题记录ID
	QuestionID         uint             `json:"question_id"`          // 题目ID
	QuestionTitle      string           `json:"question_title,omitempty"` // 题目标题（可选）
	Type               string           `json:"type,omitempty"`       // 题目类型（可选）
	Difficulty         string           `json:"difficulty,omitempty"` // 难度（可选）
	Options            []QuestionOption `json:"options,omitempty"`    // 选项列表（可选）
	UserAnswer         string           `json:"user_answer"`          // 用户答案
	CorrectAnswer      string           `json:"correct_answer,omitempty"` // 正确答案（可选）
	IsCorrect          bool             `json:"is_correct"`           // 是否正确
	Explanation        string           `json:"explanation,omitempty"` // 题目解析（可选）
	KnowledgePointID   uint             `json:"knowledge_point_id,omitempty"` // 知识点ID（可选）
	KnowledgePointName string           `json:"knowledge_point_name,omitempty"` // 知识点名称（可选）
	CreatedAt          string           `json:"created_at"`           // 创建时间
}

File: software engineering-fronted/frontend-vue-app/src/services/quiz.ts
import request from '@/utils/request'

// 题目类型
export interface Question {
  id: number
  title: string
  type: 'single' | 'multiple'
  difficulty: 'easy' | 'medium' | 'hard'
  knowledge_point_id: number
  options: Array<{ key: string; value: string }>
  answer: string
  explanation: string
  created_at: string
}

// 答题提交参数
export interface QuizSubmitParams {
  question_id: number
  user_answer: string
}

// 答题结果
export interface QuizResult {
  id: number
  question_id: number
  user_answer: string
  is_correct: boolean
  created_at: string
}

// 获取题目列表
export function getQuestions(params?: { page?: number; size?: number }) {
  return request.get<{ list: Question[]; total: number }>('/questions', { params })
}

// 获取单个题目
export function getQuestion(id: number) {
  return request.get<Question>(`/questions/${id}`)
}

// 提交答题
export function submitQuiz(data: QuizSubmitParams) {
  return request.post<QuizResult>('/quizzes/submit', data)
}

// 获取答题历史
export function getQuizHistory(params?: { page?: number; size?: number }) {
  return request.get<{ list: QuizResult[]; total: number }>('/quizzes/history', { params })
}
